runtime/src/app/app.ts

```diff
@@ -237,7 +239,7 @@ export async function hydrate_target(dest: Target): Promise<HydratedTarget> {
 
 			const { default: component, preload } = await load_component(components[part.i]);
 
-			let preloaded;
+			let preloaded: any;
```

Comment:
from the docs it sounds to me like this should be `object | Promise<object>` - https://sapper.svelte.dev/docs#Return_value

Reply:
Wow, that opened up a can of worms. The definition seems wrong; most likely it should `Object` and not `object` (a `Number` is an `object` but not an `Object`).

Reply:
Can preload return `null` / `undefined`? 🤔

Reply:
They _can_but I'm not sure what they behavior surrounding that should be - #843

Reply:
I think you might have the capitalization backwards: https://2ality.com/2020/01/typing-objects-typescript.html#object-(lowercase-%E2%80%9Co%E2%80%9D)-in-typescript%3A-non-primitive-values

Reply:
Wow, these are TypeScript subtleties I was not aware of. You're right, numbers etc are not included in `object`, so 

```
// compiles
const O: Object = 1;
// Type 'number' is not assignable to type 'object'.ts(2322)
const o: object = 1;
```

On the other hand
 
```
const O: Object = 1;
// compiles fine
const o: object = O;
```

So I think it's strictly speaking that both `object` and `Object` are assignable from `Number`, but that TS will do type coercion from `number` to `Number` only when encountering `Object`, not `object`.

Interestingly enough, this compiles too

```
const o: object = Object.create(null);
const O: Object = o;

console.log(O.toString);
```

but prints `undefined`. So `O` doesn't actually have have the `toScript` method that its type is supposed to guarantee.

I'm not sure where this leaves us. I'll switch back to `object`, though I do think it's a bit too obscure.